x-pack/plugins/triggers_actions_ui/README.md

```diff
@@ -0,0 +1,1054 @@
+  return {
+    id: 'threshold',
+    name: 'Index Threshold',
+    iconClass: 'alert',
+    alertParamsExpression: IndexThresholdAlertTypeExpression,
+    validate: validateAlertType,
+  };
+}
+```
+alertParamsExpression form represented as an expression using `EuiExpression` components:
+![Index Threshold Alert expression form](https://i.imgur.com/Ysk1ljY.png)
+
+Index Threshold Alert form with validation:
+![Index Threshold Alert validation](https://i.imgur.com/TV8c7hL.png)
+
+## Alert type model definition
+
+Each alert type should be defined as `AlertTypeModel` object with the next properties:
+```
+  id: string;
+  name: string;
+  iconClass: string;
+  validate: (alertParams: any) => ValidationResult;
+  alertParamsExpression: React.FunctionComponent<any>;
+  defaultActionMessage?: string;
+```
+|Property|Description|
+|---|---|
+|id|Alert type id. Should be the same as on the server side.|
+|name|Name of alert type, that will be displayed on the select card in UI.|
+|iconClass|Icon of alert type, that will be displayed on the select card in UI.|
+|validate|Validation function for the alert params.|
+|alertParamsExpression|React functional component for building UI of the current alert type params.|
+|defaultActionMessage|Optional property for providing default message for all added actions with `message` property.|
+
+IMPORTANT! Current UI support only one default action group. 
+Action groups is mapped from server API result for [GET /api/alert/types: List alert types](https://github.com/elastic/kibana/tree/master/x-pack/legacy/plugins/alerting#get-apialerttypes-list-alert-types).
+Server side alert type model:
+```
+export interface AlertType {
+  id: string;
+  name: string;
+  validate?: {
+    params?: { validate: (object: any) => any };
+  };
+  actionGroups: string[];
+  executor: ({ services, params, state }: AlertExecutorOptions) => Promise<State | void>;
+}
+```
+Only one default (which means first item of the array) action group is displayed in the current UI.
+Design of user interface and server API for multiple action groups is on the stage of discussion and development.
+
+## Register alert type model
+
+There are two ways of registration new alert type:
+
+1. Directly in `triggers_actions_ui` plugin. In this case alert type will be available in Create Alert flyout of the Alerts and Actions management section.
+Registration code for a new alert type model should be added to the file `x-pack/plugins/triggers_actions_ui/public/application/components/builtin_alert_types/index.ts`
+Only registered alert types are available in UI.
+
+2. Register alert type in other plugin. In this case alert type will be available only in current plugin UI. 
+It should be done by importing dependency `TriggersAndActionsUIPublicPluginSetup` and adding the next code on plugin setup:
+
+```
+function getSomeNewAlertType() {
+  return { ... } as AlertTypeModel;
+}
+
+triggers_actions_ui.alertTypeRegistry.register(getSomeNewAlertType());
+```
+
```

Comment:
It might be worth linking from here to the New Platform guide that explains how to define plugin dependencies.